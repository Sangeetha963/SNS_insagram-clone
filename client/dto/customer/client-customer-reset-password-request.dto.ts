import { ApiProperty } from '@nestjs/swagger'
import { IsEmail, IsNotEmpty } from 'class-validator'

export class ClientCustomerResetPasswordRequestDto {
    @ApiProperty()
    @IsNotEmpty()
    @IsEmail()
    email!: string
}
