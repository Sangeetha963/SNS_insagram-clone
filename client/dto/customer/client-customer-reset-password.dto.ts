import { ApiProperty } from '@nestjs/swagger'
import { IsNotEmpty, IsString, MaxLength, MinLength } from 'class-validator'

export class ClientCustomerResetPasswordDto {
    @ApiProperty({ description: '認証トークン' })
    @IsNotEmpty()
    @IsString()
    credential!: string

    @ApiProperty({ description: 'password' })
    @MinLength(6)
    @MaxLength(255)
    password!: string
}
