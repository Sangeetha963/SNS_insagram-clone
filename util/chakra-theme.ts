import { extendTheme, Theme, theme as baseTheme } from '@chakra-ui/react'

import { RecursivePartial } from '~/types/util-type'

// Chakra標準テーマの上書き用 型補完が効くようにしてある
const extendedTheme: RecursivePartial<Theme> = {
    components: {},
}

// Chakra標準テーマに無い値を追加したいときに使う
const originalTheme = {
    colors: {
        primary: '#0077C7',
        secondary: '#6FA6CC',
        white: '#FFFFFF',
        danger: '#E01E5A',
        warning: '#FF8800',
        text: {
            black: '#23221F',
            grey: '#706D65',
            link: '#0071C1',
            light: '#efefef',
        },
        toolbar: '#efefef',
    },
} as const

export const theme = extendTheme({ ...extendedTheme, ...originalTheme }) as typeof baseTheme & typeof originalTheme
