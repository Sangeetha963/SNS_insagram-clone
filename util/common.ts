import { useBreakpointValue } from '@chakra-ui/react'
import { useEffect, useLayoutEffect, useState } from 'react'

export const sleep = (ms: number): Promise<void> => {
    return new Promise<void>((resolve) => {
        setTimeout(() => {
            resolve()
        }, ms)
    })
}

export const isProd = process.env.NODE_ENV === 'production'

export const isClientSide = () =>
    typeof window !== 'undefined' &&
    typeof (window === null || window === void 0 ? void 0 : window.getComputedStyle) === 'function'

export const isServerSide = () => !isClientSide()

export const useIsomorphicLayoutEffect = typeof window !== 'undefined' ? useLayoutEffect : useEffect

export const useBreakPointValueSSR = <T>(values: Partial<Record<string, T>> | T[], defaultBreakpoint?: string): T | undefined => {
    const result = useBreakpointValue(values, defaultBreakpoint)
    const [actual, setActual] = useState<T>()

    useIsomorphicLayoutEffect(() => {
        setActual(result)
    }, [result, actual])

    return actual
}
