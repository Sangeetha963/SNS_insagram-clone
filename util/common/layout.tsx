import React, { FC, ReactNode } from 'react'

import { Default } from '~/layout/Default'

type Props = {
    children?: ReactNode
}

const LayoutComponent: FC<Props> = ({ children }) => <>{children}</>

export const convertLayoutComponent = (layout?: FC<Props> | string): FC<Props> => {
    if (typeof layout === 'function') return layout as FC
    else if (typeof layout === 'string') {
        switch (layout) {
            default:
                return Default
        }
    }

    return LayoutComponent
}
