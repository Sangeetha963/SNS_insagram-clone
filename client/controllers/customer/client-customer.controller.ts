import { Body, Controller, Delete, Get, Post, Put } from '@nestjs/common'
import { ApiBody, ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger'
import { ClientRoles } from '~/client/auth/decorator/client-role.decorator'
import { CurrentUser } from '~/util/fastify/current-user'

import { ClientAuthService } from '~/client/services/auth/client-auth.service'
import { ClientCustomerService } from '~/client/services/customer/client-customer.service'

import { CustomerEntities } from '~/entities/customer/customer.entities'

import { ClientAuthTokenResponseDto } from '~/client/dto/auth/client-auth-token-response.dto'
import { ClientCustomerChangePasswordDto } from '~/client/dto/customer/client-customer-change-password.dto'
import { ClientCustomerCreateDto } from '~/client/dto/customer/client-customer-create.dto'
import { ClientCustomerResetPasswordDto } from '~/client/dto/customer/client-customer-reset-password.dto'
import { ClientCustomerUpdateDto } from '~/client/dto/customer/client-customer-update.dto'

@Controller('client/customer')
@ApiTags('ClientCustomerController')
export class ClientCustomerController {
    constructor(readonly clientCustomerService: ClientCustomerService, readonly clientAuthService: ClientAuthService) {}

    @Get()
    @ApiOperation({ summary: 'ユーザー情報取得' })
    @ApiResponse({ type: () => CustomerEntities, status: 200 })
    @ClientRoles({ currentUser: true })
    async getUser(@CurrentUser('user') user: CustomerEntities): Promise<CustomerEntities> {
        return user
    }

    @Post()
    @ApiOperation({ summary: 'ユーザー情報作成' })
    @ApiBody({ type: () => ClientCustomerCreateDto })
    @ApiResponse({ type: () => ClientAuthTokenResponseDto, status: 200 })
    async createUser(@Body() dto: ClientCustomerCreateDto): Promise<ClientAuthTokenResponseDto> {
        const customer = await this.clientCustomerService.create(dto)
        return await this.clientAuthService.generateToken(customer)
    }

    @Put()
    @ApiOperation({ summary: 'ユーザー情報更新' })
    @ApiBody({ type: () => ClientCustomerUpdateDto })
    @ApiResponse({ type: () => CustomerEntities, status: 200 })
    @ClientRoles({ currentUser: true })
    async updateUser(
        @CurrentUser('user') customer: CustomerEntities,
        @Body() dto: ClientCustomerUpdateDto,
    ): Promise<CustomerEntities> {
        return await this.clientCustomerService.update(customer, dto)
    }

    @Post('resetPassword')
    @ApiOperation({ summary: 'パスワードリセット コード確認&パスワード変更' })
    @ApiBody({ type: () => ClientCustomerResetPasswordDto })
    @ApiResponse({ status: 200 })
    async postResetPassword(@Body() dto: ClientCustomerResetPasswordDto): Promise<void> {
        await this.clientCustomerService.resetPassword(dto)
    }

    @Put('changePassword')
    @ApiOperation({ summary: 'パスワード変更' })
    @ApiBody({ type: ClientCustomerChangePasswordDto })
    @ApiResponse({ type: CustomerEntities, status: 200 })
    @ClientRoles({ currentUser: true })
    async putChangePassword(
        @CurrentUser() customer: CustomerEntities,
        @Body() dto: ClientCustomerChangePasswordDto,
    ): Promise<CustomerEntities> {
        await this.clientCustomerService.changePassword(customer, dto)
        return customer
    }

    @Delete('')
    @ApiOperation({ summary: '退会' })
    @ClientRoles({ currentUser: true })
    async deleteWithdrawal(@CurrentUser() user: CustomerEntities): Promise<void> {
        console.log(user)
    }
    // @update('')
    // @ApiOperation({ summary: '' })
    // @ApiBody({ type: ClientCustomerUpdateDto })
    // @ApiResponse({ type: CustomerEntities, status: 200 })
    // @ClientRoles({ currentUser: true})
    // async update(
    //     @CurrentUser() customer: CustomerEntities,
    //     @Body() dto: ClientCustomerUpdateDto,
    // ): Promise<CustomerEntities> {
    //     await this.clientCustomerService.update(customer, dto)
    // }
}
