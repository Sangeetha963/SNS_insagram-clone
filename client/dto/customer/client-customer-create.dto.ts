import { OmitAndPickPartial } from '~/type/common'
import { BaseEntities } from '~/util/typeorm/base-entities'
import { ApiProperty } from '@nestjs/swagger'
import { CustomerEntities } from '~/entities/customer/customer.entities'
import { IsEmail, IsNotEmpty, IsString } from 'class-validator'

export class ClientCustomerCreateDto
    implements OmitAndPickPartial<CustomerEntities, 'auth' | keyof BaseEntities, never>
{
    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsEmail()
    email!: string

    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsNotEmpty()
    password!: string

    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsNotEmpty()
    passwordConfirm!: string

    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsNotEmpty()
    name!: string

    constructor(params: Readonly<ClientCustomerCreateDto>) {
        Object.assign(this, params)
    }
    follower!: CustomerEntities[]
    following!: CustomerEntities[]
}
