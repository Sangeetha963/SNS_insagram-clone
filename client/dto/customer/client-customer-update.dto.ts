import { OmitAndPickPartial } from '~/type/common'
import { BaseEntities } from '~/util/typeorm/base-entities'
import { ApiProperty } from '@nestjs/swagger'
import { CustomerEntities } from '~/entities/customer/customer.entities'
import { IsEmail, IsNotEmpty, IsString } from 'class-validator'

export class ClientCustomerUpdateDto
    implements OmitAndPickPartial<CustomerEntities, 'email' | 'auth' | keyof BaseEntities, never>
{
    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsEmail()
    email!: string

    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    @IsNotEmpty()
    name!: string

    constructor(params: Readonly<ClientCustomerUpdateDto>) {
        Object.assign(this, params)
    }
    follower!: CustomerEntities[]
    following!: CustomerEntities[]
}

export class ClientFollowUserDto {
    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    userId!: string
}

export class ClientUnFollowUserDto {
    @ApiProperty({ type: String, required: true, nullable: false })
    @IsString()
    userId!: string
}
