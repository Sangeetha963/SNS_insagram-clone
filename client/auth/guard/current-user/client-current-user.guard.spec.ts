import { ClientCurrentUserGuard } from './client-current-user.guard'
import { Test, TestingModule } from '@nestjs/testing'
import { getConfigServiceImpl } from '~/util/module-impl/config-module-impl'
import { UtilModule } from '~/util/util.module'
import { getJwtModuleImpl } from '~/util/module-impl/jwt-module-impl'
import { createMock } from '@golevelup/ts-jest'
import { ExecutionContext } from '@nestjs/common'
import { JwtService } from '@nestjs/jwt'
import { FastifyRequest } from 'fastify'
import { getTypeOrmTestModule } from '~/util/test/typeorm-test-module'
import { createCustomerMock } from '~/entities/customer/customer.mock'
import { JaJp } from '~/messages/ja-jp'
import { ClientCustomerService } from '~/client/services/customer/client-customer.service'
import { ClientAuthService } from '~/client/services/auth/client-auth.service'
import { ClientServicesModule } from '~/client/services/client-services.module'

const createFastifyRequestHeader = (bearer: string): Partial<FastifyRequest> => ({
    headers: {
        authorization: `bearer ${bearer}`,
    },
})

const createContextMock = (req: Partial<FastifyRequest>) =>
    createMock<ExecutionContext>({
        switchToHttp: () => ({
            getRequest: () => req,
        }),
    })

describe('ClientCurrentUserGuard', () => {
    let guard: ClientCurrentUserGuard
    let jwtService: JwtService
    let customerService: ClientCustomerService
    let clientAuthService: ClientAuthService

    beforeAll(async () => {
        const module: TestingModule = await Test.createTestingModule({
            imports: [
                getConfigServiceImpl(),
                getTypeOrmTestModule(),
                UtilModule,
                getJwtModuleImpl(),
                ClientServicesModule,
            ],
            providers: [ClientCurrentUserGuard],
        }).compile()

        jwtService = module.get(JwtService)
        guard = module.get<ClientCurrentUserGuard>(ClientCurrentUserGuard)
        customerService = module.get(ClientCustomerService)
        clientAuthService = module.get(ClientAuthService)
    })

    it('should be defined', () => {
        expect(guard).toBeDefined()
        expect(jwtService).toBeDefined()
    })

    it('should be success verify', async () => {
        jest.spyOn(customerService, 'findOneByUuid').mockImplementationOnce(async () => createCustomerMock())

        const jwtToken = await clientAuthService.signToken((await customerService.findOneByUuid(''))!)
        expect(jwtToken).toBeDefined()

        const req = createFastifyRequestHeader(jwtToken)
        const context = createContextMock(req)
        const res = await guard.canActivate(context)
        expect(res).toBe(true)
        expect(req).toMatchSnapshot({
            headers: {
                authorization: expect.any(String),
            },

            jwtPayload: {
                exp: expect.any(Number),
                iat: expect.any(Number),
                uuid: expect.any(String),
                iss: expect.any(String),
                aud: expect.any(Array),
            },
        })
    })

    it('should be failed(expired)', async () => {
        const jwtToken = jwtService.sign({ raw: 'data' }, { expiresIn: '-24h' })
        expect(jwtToken).toBeDefined()

        const req = createFastifyRequestHeader(jwtToken)
        const context = createContextMock(req)
        await expect(guard.canActivate(context)).rejects.toThrowError(JaJp()[20020101])
        expect(req.jwtPayload).toBeUndefined()
    })

    it('should be failed(invalid)', async () => {
        const jwtToken = jwtService.sign({ raw: 'data' }, { expiresIn: '-24h', secret: 'dummy_secret_token' })
        expect(jwtToken).toBeDefined()

        const req = createFastifyRequestHeader(jwtToken)
        const context = createContextMock(req)
        await expect(guard.canActivate(context)).rejects.toThrowError(JaJp()[20020102])
        expect(req.jwtPayload).toBeUndefined()
    })

    it('token is not defined', async () => {
        const context = createContextMock({
            headers: {},
        })
        await expect(guard.canActivate(context)).resolves.toBe(false)
    })
})
