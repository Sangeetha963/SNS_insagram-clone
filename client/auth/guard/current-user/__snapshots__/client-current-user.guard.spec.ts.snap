// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientCurrentUserGuard should be success verify 1`] = `
{
  "headers": {
    "authorization": Any<String>,
  },
  "jwtPayload": {
    "aud": Any<Array>,
    "exp": Any<Number>,
    "iat": Any<Number>,
    "iss": Any<String>,
    "sub": "AccessToken",
    "uuid": Any<String>,
  },
}
`;
